File: demo_v2_threshold/demo_v2/backend/app/schemas/timeline.py
"""Timeline and trend schemas."""

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


class DimensionHistoryRow(BaseModel):
    dimension_name: str
    score: float


class HealthHistoryEvent(BaseModel):
    submission_id: UUID
    date: datetime
    overall_score: float | None
    rag_status: str | None
    dimensions: list[DimensionHistoryRow]


class SubmissionTimelineEvent(BaseModel):
    submission_id: UUID
    date: datetime
    overall_score: float | None
    rag_status: str | None
    status_code: str
    trend: Literal["improving", "declining", "stable", "none"]
    action_description: str
    actor_name: str | None
    actor_role: str | None
    
    # lifecycle dates
    created_at: datetime
    submission_date: datetime | None
    approval_date: datetime | None
    locked_at: datetime | None

File: demo_v2_threshold/demo_v2/frontend/src/types/timeline.ts
export interface SubmissionTimelineEvent {
  submission_id: string;
  date: string;
  overall_score: number | null;
  rag_status: string | null;
  status_code: string;
  trend: "improving" | "declining" | "stable" | "none";
  action_description: string;
  actor_name: string | null;
  actor_role: string | null;
  created_at: string;
  submission_date: string | null;
  approval_date: string | null;
  locked_at: string | null;
}

export interface DimensionHistoryRow {
  dimension_name: string;
  score: number;
}

export interface HealthHistoryEvent {
  submission_id: string;
  date: string;
  overall_score: number | null;
  rag_status: string | null;
  dimensions: DimensionHistoryRow[];
}
